src/blocking/spi.rs

```diff
@@ -108,3 +108,134 @@ pub mod write_iter {
         }
     }
 }
+
+/// Provides SpiWithCS wrapper around an spi::* and OutputPin impl
+pub mod spi_with_cs {
+
+    use core::marker::PhantomData;
+    use core::fmt::Debug;
+
+    use crate::digital::OutputPin;
+    use super::*;
+
+    /// SpiWithCS wraps an blocking::spi* implementation with Chip Select (CS)
+    /// pin management.
+    /// For sharing SPI between peripherals, see [shared-bus]()
+    pub struct SpiWithCs<Spi, SpiError, Pin, PinError> {
+        spi: Spi,
+        cs: Pin,
+
+        _spi_err: PhantomData<SpiError>,
+        _pin_err: PhantomData<PinError>,
+    }
+
+    /// SpiWithCsErr provies an error numberation over generic Spi and Pin variants
+    #[derive(Clone, Debug, PartialEq)]
+    pub enum SpiWithCsErr<SpiError, PinError> {
+        /// Underlying SPI error
+        Spi(SpiError),
+        /// Underlying Pin error
+        Pin(PinError),
+    }
+
+    /// ManagedCS marker trait indicates Chip Select management is automatic
+    impl <Spi, SpiError, Pin, PinError> ManagedCs for SpiWithCs <Spi, SpiError, Pin, PinError> {}
+    
+    impl <Spi, SpiError, Pin, PinError> SpiWithCs <Spi, SpiError, Pin, PinError> 
+    where
+        Pin: crate::digital::OutputPin<Error=PinError>,
+        SpiError: Debug,
+        PinError: Debug,
+    {
+        /// Create a new SpiWithCS wrapper with the provided Spi and Pin
+        pub fn new(spi: Spi, cs: Pin) -> Self {
+            Self{spi, cs, _spi_err: PhantomData, _pin_err: PhantomData}
+        }
+
+        /// Fetch references to the inner Spi and Pin types.
+        /// Note that using these directly will violate the `ManagedCs` constraint.
+        pub fn inner(&mut self) -> (&mut Spi, &mut Pin) {
+            (&mut self.spi, &mut self.cs)
+        }
+    }
+
+    impl <Spi, SpiError, Pin, PinError> Transfer<u8> for SpiWithCs<Spi, SpiError, Pin, PinError> 
+    where 
+        Spi: Transfer<u8, Error=SpiError>,
+        Pin: OutputPin<Error=PinError>,
+        SpiError: Debug,
+        PinError: Debug,
+    {
+        type Error = SpiWithCsErr<SpiError, PinError>;
+
+        /// Attempt an SPI transfer with automated CS assert/deassert
+        fn try_transfer<'w>(&mut self, data: &'w mut [u8]) -> Result<&'w [u8], Self::Error> {
+            // First assert CS
+            self.cs.try_set_low().map_err(SpiWithCsErr::Pin)?;
+
+            // Attempt the transfer, storing the result for later
+            let spi_res = self.spi.try_transfer(data).map_err(SpiWithCsErr::Spi);
```

Comment:
Can we call this just `result`?
```suggestion
            let result = self.spi.try_transfer(data).map_err(SpiWithCsErr::Spi);
```

Reply:
is result more clear?

Reply:
I prefer whole words, yes, but feel free to dismiss this.